test/unit/alignment/matrix/detail/alignment_score_matrix_one_column_banded_test.cpp

```diff
@@ -0,0 +1,107 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <vector>
+#include <string>
+
+#include <seqan3/alignment/matrix/detail/alignment_score_matrix_one_column_banded.hpp>
+
+#include "alignment_matrix_base_test_template.hpp"
+#include "simulated_alignment_test_template.hpp"
+#include "../../../range/iterator_test_template.hpp"
+
+using namespace seqan3;
+
+template <typename t>
+struct alignment_score_matrix_one_column_banded_test
+{
+    using matrix_t = detail::alignment_score_matrix_one_column_banded<t>;
+    using score_type = t;
+
+    alignment_score_matrix_one_column_banded_test() = default;
+    alignment_score_matrix_one_column_banded_test(std::string f, std::string s) :
+        matrix{matrix_t{f, s, static_band{lower_bound{-2}, upper_bound{2}}, -100}}
+    {}
+
+    // Banded matrix. We write only partial columns to the result.
+    // The remaining part is therefor 0.
+    std::vector<t> gold_matrix{  0, -1, -2,
+                                -1, -1, -1, -2,
+                                -2, -1, -2, -1, -2,
+                                    -2, -2, -2, -2,
+                                        -2, -3, -2,
+                                 0, 0, 0, 0, 0, 0};
+
+    matrix_t matrix{};
+    size_t last_init_column = 2;
+};
+
+INSTANTIATE_TYPED_TEST_CASE_P(one_column_banded,
+                              simulated_alignment_test,
+                              alignment_score_matrix_one_column_banded_test<int32_t>);
+
+using test_type = std::pair<detail::alignment_score_matrix_one_column_banded<int32_t>, std::true_type>;
+
+INSTANTIATE_TYPED_TEST_CASE_P(one_column_banded,
+                              alignment_matrix_base_test,
+                              test_type);
+
+//-----------------------------------------------------------------------------
+// Test outer iterator
+//-----------------------------------------------------------------------------
+struct outer_iterator
+{};
+
+template <>
+struct iterator_fixture<outer_iterator> : alignment_matrix_base_test<test_type>
```

Comment:
why do you create an extra tag if you can just use `alignment_matrix_base_test<test_type>` as the type specialisation?

Reply:
Because the current design does not allow me to define another iterator_fixture with the same test type (see below), because I would define the same type twice. One of the points I wasn't happy with in the first place.

Reply:
Well because you should templatize the fixture over the iterator type that you are testing, not the range type. And your iterator types are not the same. 
In the first case it is `decltype(alignment_matrix_base_test<test_type>::test_range.begin())`  (which I am sure can be written out more nicely.). That way the test directly displays the actual type. 
and in the second case it is `decltype((*alignment_matrix_base_test<test_type>::test_range.begin()).begin())` 
But I don't care you think this artificial struct is nicer.

Reply:
> Well because you should templatize the fixture over the iterator type that you are testing, not the range type :D. And your iterator types are not the same.

But that's exactly the problem that I was referring to. I usually got the iterator from the range not the other way around. So I would need to define the types globally then get the iterators from them and then implement some fixture base classes that kind of infer back from the iterator type to the range type to provide the actual data. And now, since I have two different flavors each of the them with two different iterator types this all makes it a little confusing about how you apply your types. So it feels like the spezialisation of the iterator_fixture is kind of weird since I only have to spezialise it to tell gtest that this is the correct fixture to use. Anyway, other people might see it differently and I found a way to get what I want (with probably the same loc as in my suggested version) so all good from my side.